### DISCUSSION TITLE
How to resize image in Quarto revealjs presentation

### DISCUSSION THREAD

**Discussion by dmkaplan2000** [quarto-dev/quarto-cli#5701] [Q&A] [answered]
I have been trying to resize an image I have placed inside a Quarto presentation and the standard method of doing so seems to have no effect on image size. Am I doing something wrong? Is there another approach?

For example, in the following presentation, the `{width=10px}` has no effect:

```
---
title: "Untitled"
format: revealjs
---

## Smaller image?

![ ](image.png){width=10px}
```

I have tried different units for the width (%, px, in) and with and without quotes, nothing seems to have any effect. The only thing that seems to work is putting the image inside a column. Even if that column is of width 100%, then resizing of the image works. For example, the following works fine, the image covers 50% of the screen width:

```
---
title: "Untitled"
format: revealjs
---

## Smaller image?

:::: {.columns}

::: {.column width="100%"}
![](images/pons.et.al.submitted.png){width="50%"}
:::

::::
```

But this seems like a strange solution. Is there a better way?

Thanks,
David

**Comment by mcanouil** [comment, marked as answer]
By default, figures are stretched, meaning the width/height are set by this feature.

See two ways to make it work.

<table style="width:100%">
<tr><th style="width:50%">Option 1</th><th style="width:50%">Option 2</th></tr>
<tr>
<td>

````qmd
---
title: "Reproducible Quarto Document"
format: revealjs
---

## Introduction {.nostretch}

![](elephant.png){width="10%"}

````

</td>
<td>

````qmd
---
title: "Reproducible Quarto Document"
format:
  revealjs:
    auto-stretch: false
---

## Introduction

![](elephant.png){width="10%"}

````

</td>
</tr>
</table>

EDIT: see <https://quarto.org/docs/presentations/revealjs/advanced.html#stretch>.

**Reply by dmkaplan2000** [reply]
Thanks, this clears it up for me.

**Reply by dmkaplan2000** [reply]
One thing though: Why does it work when I add a column? Wouldn't the image be stretched within the column?

**Reply by mcanouil** [reply]
Because it's no longer a "plain" figure, thus the auto-strech is not applied.

**Reply by dmkaplan2000** [reply]
Thanks!

**Reply by cderv** [reply]
For reference to anyone looking for this, doc about this revealjs stretch feature: https://quarto.org/docs/presentations/revealjs/advanced.html#stretch

**Comment by dmkaplan2000** [comment]
Thanks. The cited text would be a bit clearer if it said:

> `auto-stretch` will only be applied to non-nested images, i.e., images that are not contained within feature blocks (e.g fragments, layout panel, columns, … ) or custom `div`'s

Also, for things like `columns` it is not clear if you can opt in to `auto-stretch` (the text talks about custom divs, but not feature blocks).

**Reply by mcanouil** [reply]
Use the `.r-stretch` class as showed in the documentation.

I don't get your point about the clearer part of the documentation.
PRs are welcome.

**Comment by dmkaplan2000** [comment]
I am just suggesting that the text itself of the last part of the stretch section could be improved to enhance clarity. But I get it. Thanks for the help.
